Add lsblk to "stratis pool report"

bgurney-rh opened the pull request:
Related: #1000

mulkieran commented:
@bgurney-rh Be sure to rebase in order to bring in recent changes.

mulkieran commented on src/stratis_cli/_actions/_report_pool.py:
```diff
@@ -0,0 +1,160 @@
+        )
+
+        # replace the lines below with the correct call to lsblk
+        blkdevs = [dev.Devnode() for dev in modevs]
```
Should we sanitize the ```blkdevs``` argument in any way? Given that lsblk will just print out data and that the values are read off the stratisd D-Bus we probably don't need to worry about their validity. Should we limit their number to 5 or 10 or something like that, so as not to be overwhelmed with output?

mulkieran reviewed (changes requested):
Please rebase once again, I had to do a Makefile fix. Also, the decision we make regarding sanitizing input to lsblk here will probably apply to stopped pools also.

mulkieran commented on src/stratis_cli/_actions/_report_pool.py:
```diff
@@ -0,0 +1,160 @@
+
+        # replace the lines below with the correct call to lsblk
+        blkdevs = [dev.Devnode() for dev in modevs]
+        subprocess.run(["lsblk", "-i"] + blkdevs, check=True)
```
For this line bandit reports two errors: B603 and B607. We should either explicitly allow these using the ```nosec``` annotation _and_ a comment explaining why _or_ correct the code so that bandit doesn't report any error.

bgurney-rh replied:
Resolved; the B603 error is ignored due to the inputs coming from D-Bus.

mulkieran commented on src/stratis_cli/_actions/_report_pool.py:
```diff
@@ -154,7 +155,8 @@ def selection_func(_uuid, info):
 
-        # Substitute lsblk call here
+        # Ignore bandit B603 errors.  Input comes from D-Bus and has
+        # been processed.
```
Try to get a single list of devnodes, just as you did with the started pools. You should be able to use roughly the same style of list comprehension as you did there, and eliminate the ```for-loop``` entirely.

bgurney-rh replied:
Resolved.

mulkieran commented on src/stratis_cli/_actions/_report_pool.py:
```diff
@@ -0,0 +1,166 @@
+        # Ignore bandit B603 errors.  Input comes from D-Bus and has
+        # been processed.
+        blkdevs = [dev[dbus.String("devnode")] for dev in stopped_pool["devs"]]
+        for dev in stopped_pool["devs"]:
```
```for-loop``` should be eliminated entirely.

bgurney-rh replied:
Resolved.

mulkieran reviewed (commented):
We need to abstract over the shared code in list and report modules before we proceed with this directly. There are several things that need abstracting:
* how we interact with the StoppedPools property
* how we manage the pool selection can be abstracted further
* how we handle the UUID formatting, possibly

I'm putting this back in pending for that reason and will think about the StoppedPools property particularly.

mulkieran commented:
@bgurney-rh Plz rebase and resolve conflicts.

bgurney-rh commented:
Rebased; the conflict was from `from ._utils import get_clevis_info` changing to `from ._utils import ClevisInfo`.

fsuba commented:
/packit/build

fsuba commented:
/packit/ build

fsuba commented:
/packit build

bgurney-rh commented:
/packit build

fsuba commented:
/packit build

bgurney-rh commented:
Rebased, resolved conflicts, and fixed a linting error on this import statement:
```
from .._stratisd_constants import PoolIdType
```
```
************* Module stratis_cli._actions._report_pool
src/stratis_cli/_actions/_report_pool.py:26:0: E0611: No name 'PoolIdType' in module 'stratis_cli._stratisd_constants' (no-name-in-module)
```

bgurney-rh commented:
/packit build
